sdk/python/approzium/opentelemetry/__init__.py

```diff
@@ -0,0 +1,51 @@
+import opentelemetry.ext.dbapi
+from opentelemetry.ext.dbapi import DatabaseApiIntegration
+
+original_DatabaseApiIntegration = None
+
+
+class _ApproziumDatabaseApiIntegration(DatabaseApiIntegration):
+    def get_connection_attributes(self, connection):
+        super().get_connection_attributes(connection)
+        # check if this is an Approzium connection
+        if hasattr(connection, "authenticator"):
+            self.add_approzium_span_attributes(connection)
+
+    def add_approzium_span_attributes(self, connection):
+        for key, value in connection.authenticator.attribution_info.items():
+            self.span_attributes["approzium." + key] = value
+
+
+def instrument():
```

Comment:
Question, should we require the caller to _call_ instrument? Or should we just automatically start up tracing? If we automatically started it, we could have a default tracing configuration, but allow it to be overridden with environment variables.

Reply:
Good question :) We could have it happen on import, or even as a silent thing when a connection is instantiated, so that it happens automatically. I am not sure what "tracing configuration" refers to. Can you elaborate?

Reply:
Interesting! On one hand, this would be great in that it's more convenient and trace by default would strengthen our observability stance. OTOH, explicitly calling approzium's `instrument()` seems to be more aligned with the typical behavior of OpenTelemetry integrations (see [pymysql integration here](https://opentelemetry-python.readthedocs.io/en/stable/ext/pymysql/pymysql.html)). I think that there's value in aligning with the typical flow in which the user might expect to see all the trace sources in their code.

Reply:
By "the tracing configuration", I mean basically [this line of code](https://github.com/cyralinc/approzium/pull/123/files#diff-5932bb857cbe1a57c9783821df9d2274R20) in our example. I'm thinking of a situation where instead of putting the responsibility onto our SDK user to do that instantiation, we would do it ourselves. 

The Jaeger Go client can be configured by [a bunch of environment variables](https://github.com/jaegertracing/jaeger-client-go#environment-variables) and the way I was thinking of doing it was:

- We emit tracing by default, and we don't really care whether people want to consume it, it's just there
- If they don't set any environment variables, we use a default config like the one you've set on that line I linked above
- If the _do_ set environment variables, we override our default configuration.

I'm not sure this is _the_ best way to do things, I'm still toying with it and I merely suggest this as something we might consider.

And regarding taking examples from others, yes, definitely there's value in that. I _do_ think there's a question as to whether _our SDK user_ should be the one doing that, or _we_ should be the ones doing that for them. One thing I wonder is, if they have to do all this calling anyways, why are we building it into the SDK rather than just letting them directly do it?

Like I said, I'm still getting my head around all this tracing stuff so there's definitely stuff I'm probably missing.